### PR TITLE
Forward-port to work with 0.11.7

### PR DESCRIPTION
**Pull request by damncabbage**
(Makes use of purescript-retro-foreign.)

! @sphvn @tmcgilchrist @charleso 
/jury approved @charleso @tmcgilchrist

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -20,15 +20,15 @@
...
-    "purescript-js-date": "^2.0.0",
-    "purescript-media-types": "^1.0.0",
-    "purescript-nullable": "^1.0.0",
-    "purescript-prelude": "^1.0.0",
-    "purescript-unsafe-coerce": "^1.0.0"
+    "purescript-datetime": "^3.4.1",
+    "purescript-enums": "^3.2.1",
+    "purescript-exceptions": "^3.1.0",
+    "purescript-foldable-traversable": "^3.6.1",
+    "purescript-retro-foreign": "git://github.com/ambiata/purescript-retro-foreign.git#topic/forward-port-for-0117",
```
Comment (tmcgilchrist) [inline bower.json]: Should this point to master rather than a branch?

**Comment by charleso** [comment]
👍

File: loom.toml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    ]
+
+[purs.test]
+  paths = [
+      "test"
+    ]
+  main = "Test.Main"
+
+[purs.test.dependencies]
+  # TODO
```
Comment (tmcgilchrist) [inline loom.toml]: un-todo this?

File: test/Main.purs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module Test.Main where
+
+import Prelude
+
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE, log)
+
+main :: Eff (console :: CONSOLE) Unit
+main = do
+  log "TODO: Tests"
```
Comment (tmcgilchrist) [inline test/Main.purs]: Tests?

**Comment by tmcgilchrist** [comment]
👍 

Note the renaming (as explained by Rob)
```
 * ---> Type
 ! ---> Effect
```

File: bower.json (hunk 1 above)
Comment (damncabbage) [reply bower.json]: oh bugger; yes

(The build is using `loom` to do the installation, as will Manor, but it should be in sync, yeah.)

File: loom.toml (hunk 2 above)
Comment (damncabbage) [reply loom.toml]: Well, I still need to figure out which dependencies (top-level and children) are only used for tests, eg. purescript-console. Keeping the TODO for now.

File: test/Main.purs (hunk 3 above)
Comment (damncabbage) [reply test/Main.purs]: There were none in the library. This is a temporary fork, until we can get to using the real/up-to-date one; at which point we can use the library directly.

(The up-to-date library actually has tests, but then we'd need a browser environment to run them [a can of worms], which makes me loathe to backport them. After we ditch this fork, it's kinda not our problem anymore.)

File: bower.json (hunk 1 above)
Comment (damncabbage) [reply bower.json]: Fixed.

**Pull request closed without merging**